src/Storage/StreamWrapper.php

```diff
@@ -0,0 +1,612 @@
+        return $this->makeStatArray([
+            'mode'    => $mode,
+            'size'    => $this->stream->getSize()
+        ]);
+    }
+
+    /**
+     * Callback handler for checking to see if the stream is at the end of file.
+     *
+     * @return bool
+     */
+    public function stream_eof()
+    {
+        return $this->stream->eof();
+    }
+
+    /**
+     * Callback handler for trying to close the stream.
+     */
+    public function stream_close()
+    {
+        if (isset($this->stream)) {
+            $this->stream->close();
+        }
+    }
+
+    /**
+     * Callback handler for trying to seek to a certain location in the stream.
+     *
+     * @param int $offset The stream offset to seek to
+     * @param int $whence Flag for what the offset is relative to. See:
+     *        http://php.net/manual/en/streamwrapper.stream-seek.php
+     * @return bool
+     */
+    public function stream_seek($offset, $whence = SEEK_SET)
+    {
+        if ($this->stream->isSeekable()) {
+            $this->stream->seek($offset, $whence);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Callhack handler for inspecting our current position in the stream
+     *
+     * @return int
+     */
+    public function stream_tell()
+    {
+        return $this->stream->tell();
+    }
+
+    /**
+     * Callback handler for trying to close an opened directory.
+     *
+     * @return bool
+     */
+    public function dir_closedir()
+    {
+        return false;
+    }
+
+    /**
+     * Callback handler for trying to open a directory.
+     *
+     * @param string $path The url directory to open
+     * @param int $options Whether or not to enforce safe_mode
+     * @return bool
+     */
+    public function dir_opendir($path, $options)
+    {
+        $this->openPath($path);
+        $this->dir_rewinddir();
+        return true;
+    }
+
+    /**
+     * Callback handler for reading an entry from a directory handle.
+     *
+     * @return string
+     */
+    public function dir_readdir()
+    {
+        $object = $this->directoryGenerator->current();
+        if ($object) {
+            $this->directoryGenerator->next();
+            return $object->name();
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Callback handler for rewind the directory handle.
+     *
+     * @return bool
+     */
+    public function dir_rewinddir()
+    {
+        $this->directoryGenerator = $this->bucket->objects([
+            'prefix' => $this->file,
+            'fields' => 'items/name,nextPageToken'
+        ]);
+        return true;
+    }
+
+    /**
+     * Callback handler for trying to create a directory.
+     *
+     * @param string $path The url directory to creaet
+     * @param int $mode The permissions on the directory
+     * @param int $options Bitwise mask of options
+     * @return bool
+     */
+    public function mkdir($path, $mode, $options)
```

Comment:
Currently it does not support either creating a bucket or using mode to set ACLs. I will investigate.